ca/certificates.go

```diff
@@ -185,11 +187,22 @@ func (rca *RootCA) RequestAndSaveNewCertificates(ctx context.Context, kw KeyWrit
 	if err != nil {
 		return nil, err
 	}
-	// Include our current root pool
+	// We retrieve the certificate with the current root pool, so we know this was issued by a legitimate manager.
+	// However, there might have been a server-side root rotation, so we verify this cert with a new pool.
+	// If we got a valid rootCABundle, turn it into a Pool, and verify the newly minted certificate using it.
+	rootCAPool := rca.Pool
+	newRootCA, newRootErr := NewRootCA(rootCABundle, nil, rca.Path, time.Minute)
```

Comment:
Also, is it possible to skip all this, particularly saving the root CA cert to disk, if the received bundle is identical to what we already have?